blocks/url-input/index.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import './style.scss';
+import { __ } from 'i18n';
+import { Component } from 'element';
+import { IconButton } from 'components';
+
+class UrlInput extends Component {
+	constructor() {
+		super( ...arguments );
+		this.toggle = this.toggle.bind( this );
+		this.submitLink = this.submitLink.bind( this );
+		this.state = {
+			expanded: false,
+		};
+	}
+
+	toggle() {
+		this.setState( { expanded: ! this.state.expanded } );
+	}
+
+	submitLink( event ) {
+		event.preventDefault();
+		this.toggle();
+	}
+
+	render() {
+		const { url, onChange } = this.props;
+		const { expanded } = this.state;
+
+		return (
+			<li className="components-url-input">
+				<IconButton
+					icon="admin-links"
+					onClick={ this.toggle }
+					className={ classnames( 'components-toolbar__control', {
+						'is-active': url,
+					} ) }
+				/>
+				{ expanded &&
+					<form
+						className="editable-format-toolbar__link-modal"
+						onSubmit={ ( event ) => this.submitLink( event ) }>
+						<IconButton className="components-url-input__back" icon="arrow-left-alt" onClick={ this.toggle } />
+						<input
+							className="editable-format-toolbar__link-input"
+							type="url"
+							required
```

Comment:
How will this work when trying to un-set the URL from the image? Currently, it shows an error that the field is required, but simply toggling "back" will show that it's truly removed (only because we're updating the block attribute `onChange` and not `onSubmit`).